### PR TITLE
PLAT-10847: Wrote migration guide from BDK 1.x to BDK 2.0

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
PLAT-10847

### Description
Write a migration guide following what has been done for the Java BDK: https://symphony-bdk-java.finos.org/migration.html

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [N/A] Unit tests updated or added
- [N/A] Docstrings added or updated
- [x] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: docsrc/markdown/migration.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    "podHost": "acme.symphony.com",
+    "agentHost": "acme.symphony.com",
+    "botUsername": "bot-username",
+    "botPrivateKeyPath": "/folder/to/private/key/",
+    "botPrivateKeyName": "rsa-privatekey.pem",
+    "truststorePath": ""
+}
+```
+
+### Python BDK 2.0
```
Comment (symphony-youri) [inline docsrc/markdown/migration.md]: should it be 4 # (indentation level)?

Comment (symphony-youri) [reply docsrc/markdown/migration.md]: and perhaps keep the same title style for before/after 
below it is named _Using the BDK X.X_ (I would use this one)

File: docsrc/markdown/migration.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+class PingPongListener(RealTimeEventListener):
+
+    def __init__(self, message_service):
+        self._message_service = message_service
+
+    async def on_message_sent(self, initiator: V4Initiator, event: V4MessageSent):
+        message_text = get_text_content_from_message(event.message)
+        stream_id = event.message.stream.stream_id
+        if message_text == "/ping":
+            await self._message_service.send_message(stream_id=stream_id, message="<messageML>pong</messageML>")
```
Comment (symphony-youri) [inline docsrc/markdown/migration.md]: maybe we can drop the messageml tags? (or perhaps you wanted to be close from the 1.x example)

**Review by symphony-youri — COMMENTED** [review]
I was looking at the legacy readme and it also mentions a SymElementsParser 
 sym_api_client_python/processors/sym_elements_parser.py
Should we say that we don't have a replacement for that (or that you should use this class for the generated models)?

File: docsrc/markdown/migration.md (hunk 2 above)
Comment (symphony-elias) [reply docsrc/markdown/migration.md]: Yes indeed, we can remove them in both cases

**Review by symphony-youri — APPROVED** [review]

**Pull request merged**